fix for issue #1742: corrected run.js, run.test.js, watch.js for the use case runOnChangeOnly=true

MonarchChakri opened the pull request:
Updated

1. test/monitor/run.test.js: to run the test which was previously skipped.
2. lib/monitor/run.js: to handle the start on change only case (-C/--on-change-only) by adding an early exit.
3. lib/monitor/watch.js: to precheck config.options.execOptions before accessing script.

MonarchChakri commented:
@remy Did you get time to look at this PR?

remy commented:
Caught me at a personal time that I was actively avoiding the computer. I must have looked at it once, but I couldn't see tests for such significant changes. Happy to reopen if there's checks for the specific changes (that either shows fixes have been applied or otherwise).

MonarchChakri commented:
@remy
This commit was given to fix the issue https://github.com/remy/nodemon/issues/1742: --on-change-only / -C options were not working and it was marked as a bug.
Found the root cause of the issue and fixed it in this commit.

Updated the test case to be run now in **`test/monitor/run.test.js`** to run the test which was previously skipped.
Also it was not caught by the tests because the test was never actually executed under test/monitor/run.test.js.
Please review and merge.

![image](https://user-images.githubusercontent.com/23417550/95023100-bc5ded80-0698-11eb-87d9-2f660fa198e0.png)

MonarchChakri commented:
Please verify the test case for runOnChangeOnly under **`test/monitor/run.test.js`**

remy commented:
Okay, I'm always super wary of large code changes, even if they're rearranging, but this change still has all tests passing so either the 100s of tests I've written are garbage or your code hasn't broken anything 👍 

Merging now.

MonarchChakri commented:
Thanks for merging and adding the label **`hacktoberfest-accepted`**. 😄 
But as per the rules I need to submit a PR during October and I have submitted it long back.
Maybe assign me another issue which I can work on it or I'll try next time. 😅
